libraries/Microsoft.Bot.Builder.Ai/SmartTalkMiddleware.cs

```diff
@@ -0,0 +1,56 @@
+﻿using Newtonsoft.Json;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Net.Http;
+using System.Threading.Tasks;
+using Microsoft.Bot.Schema;
+
+namespace Microsoft.Bot.Builder.Ai
+{
+    public class SmartTalkMiddleware : IMiddleware
+    {
+        private readonly SmartTalk _smartTalk;
+        private readonly SmartTalkMiddlewareOptions _options;
+
+        public SmartTalkMiddleware(SmartTalkMiddlewareOptions options, HttpClient httpClient = null)
+        {
+            _options = options ?? throw new ArgumentNullException(nameof(options));
+
+            _smartTalk = new SmartTalk(options, httpClient);
+        }
+
+        public async Task OnTurn(ITurnContext context, MiddlewareSet.NextDelegate next)
```

Comment:
This will run on EVERY SINGLE message that hits the bot. 

This means every message will be bounced off of your SmartTalk server, introducting significant latency even for cases where your code probably isn't relevant. 

IS this the behavior you want?